### PR TITLE
PYIC-6978 Updated configService featureFlag enabled check, log and return false in case of exception from ssm.

### PR DESCRIPTION
**Pull request by rpayal**
## Proposed changes
Handle exception from SSM for feature flag enabled check. Return false in case of featureFlag missing for the env.

### What changed
Updated configService featureFlag enabled check.
- log warning message.
- return false in case of featureFlag missing for the env.

### Issue tracking

- [PYIC-6978](https://govukverify.atlassian.net/browse/PYIC-6978)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-6978]: https://govukverify.atlassian.net/browse/PYIC-6978?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 1]
```diff
@@ -265,13 +266,24 @@ public Map<String, List<MitigationRoute>> getCimitConfig() throws ConfigExceptio
...
+                            getSsmParameter(ConfigurationVariable.FEATURE_FLAGS, featureFlagValue));
+        } catch (SsmException ex) {
+            LOGGER.warn(
+                    (new StringMapMessage())
+                            .with(
+                                    LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                    "SSM parameter not found for feature flag:"
+                                            + featureFlagValue));
+        }
+        return enabled;
```
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: minor thing, but I think it'd be clearer to get rid of the variable and explicitly `return false` inside the catch

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java [hunk 2]
```diff
@@ -265,13 +266,24 @@ public Map<String, List<MitigationRoute>> getCimitConfig() throws ConfigExceptio
...
 
     public boolean enabled(String featureFlagValue) {
-        return Boolean.parseBoolean(
-                getSsmParameter(ConfigurationVariable.FEATURE_FLAGS, featureFlagValue));
+        boolean enabled = false;
+        try {
+            enabled =
+                    Boolean.parseBoolean(
+                            getSsmParameter(ConfigurationVariable.FEATURE_FLAGS, featureFlagValue));
+        } catch (SsmException ex) {
```
Comment (Joe-Edwards-GDS) [inline libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: Is there anything more precise we can/should filter on here (i.e. so we only catch 'parameter not found' errors, and not other exceptions)?

Comment (rpayal) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: yeah thought so.....can use
ssmProvider.get - ParameterNotFoundException 
ssmProvider.getMultiple - InvalidFilterKeyException 

should be ok just to use ParameterNotFoundException.

Comment (rpayal) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: done

File: libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java (hunk 1 above)
Comment (rpayal) [reply libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/ConfigService.java]: done

**Review by Joe-Edwards-GDS — APPROVED** [review]

**Pull request merged**